### PR TITLE
tests for user commands

### LINKED ISSUE #133: Improve tests for user commands
Right now, we have no tests for the `execute()` methods of user commands, and they have a lower coverage than most of the code.

### PR DESCRIPTION
**Pull request by M1keReck**
fixes #133

### CODE REVIEW

File: src/ja/server/database/database.py [hunk 1]
```diff
@@ -20,15 +20,6 @@ def find_job_by_id(self, job_id: str) -> DatabaseJobEntry:
         @return The stored data about the job, or None if no such job was found.
         """
 
-    @abstractmethod
```
Comment (ammen99) [inline src/ja/server/database/database.py]: Why did you remove `find_job_by_label`, and then you reimplement it using `query_jobs` and manually filter through them?

File: src/test/user/cli.py [hunk 2]
```diff
@@ -46,70 +48,106 @@ def setUp(self) -> None:
...
         command: QueryCommand = cast(QueryCommand, self._cli.get_server_command("query --sr lic,lic1".split()))
         response: str = self._proxy.query(command).result_string
+        response1: str = command.execute(self._db).result_string
         self.assertEqual(response, "\n".join([str(self._jobs[4])]))
+        self.assertEqual(response1, "\n".join([str(self._jobs[4])]))
 
     def test_cancel_label(self) -> None:
         command: CancelCommand = cast(CancelCommand, self._cli.get_server_command("cancel --label lab".split()))
         self._proxy.cancel_job(command)
+        command.execute(self._db).result_string
```
Comment (nikolatzotchev) [inline src/test/user/cli.py]: you forgot to remove .result_string (or forgot to assign it)

File: src/test/user/cli.py [hunk 3]
```diff
@@ -46,70 +48,106 @@ def setUp(self) -> None:
...
+        self.assertEqual(response1, "\n".join([str(self._jobs[1])]))
 
     def test_query_memory_empty(self) -> None:
         command: QueryCommand = cast(QueryCommand, self._cli.get_server_command("query --memory 10 40".split()))
         response: str = self._proxy.query(command).result_string
+        response1: str = command.execute(self._db).result_string
         self.assertEqual(response, "No jobs satisfy these constraints.")
+        self.assertEqual(response1, "No jobs satisfy these constraints.")
 
     def test_query_sresources(self) -> None:
```
Comment (nikolatzotchev) [inline src/test/user/cli.py]: sresources -> resources

**Comment by ammen99** [comment]
I think we have missed this when doing the previous reviews. The tests for commands should be separate, not part of the CLI tests.

File: src/ja/server/database/database.py (hunk 1 above)
Comment (M1keReck) [reply src/ja/server/database/database.py]: Because I couldn't make it work with `session.query(Job).filter(Job.label == label)`. The method didn't work.

File: src/test/user/cli.py (hunk 3 above)
Comment (M1keReck) [reply src/test/user/cli.py]: That stands for special resources.

File: src/ja/server/database/database.py (hunk 1 above)
Comment (JohannesGaessler) [reply src/ja/server/database/database.py]: I opened a PR that fixes the method: https://github.com/DistributedTaskScheduling/JobAdder/pull/135

Comment (ammen99) [reply src/ja/server/database/database.py]: That PR is now merged, so it makes the most sense to use the database function.

File: src/ja/server/database/sql/database.py [hunk 4]
```diff
@@ -196,6 +196,7 @@ def update_job(self, job: Job) -> None:
                                                                     0, 0),
                                          machine=None)
             session.add(job_entry)
+            # job.uid =
```
Comment (M1keReck) [inline src/ja/server/database/sql/database.py]: How could I get the id of the newly added job entry before adding it? Since the id has to be assigned to the job before adding it to the DB

Comment (nikolatzotchev) [reply src/ja/server/database/sql/database.py]: the id is assigned automatically, uid is not, you can't get the id before adding.

**Comment by ammen99** [comment]
@M1keReck It is generally better to rebase on top of master instead of merging branches.

**Comment by ammen99** [comment]
I manually re-triggered the TravisCI build because I suspected we'll have failing tests.. However I think that maybe you just need to rebase on top of master.
